awx/main/tests/functional/test_jobs.py

```diff
@@ -46,6 +46,13 @@ def test_job_notification_data(inventory, machine_credential, project):
     notification_data = job.notification_data(block=0)
     assert json.loads(notification_data['extra_vars'])['SSN'] == encrypted_str
 
+@pytest.mark.django_db
+def test_job_notification_host_data(inventory, machine_credential, project, job_template, host):
+    job = Job.objects.create(
+        job_template=job_template, inventory=inventory, name='hi world', project=project
+    )
+    JobHostSummary.objects.create(job=job, host=host, changed=1, dark=2, failures=3, ok=4, processed=3, skipped=2, rescued=1, ignored=0)
+    assert job.notification_data()['hosts'] == {'single-host': {'failed': True, 'changed': 1, 'dark': 2, 'failures': 3, 'ok': 4, 'processed': 3, 'skipped': 2, 'rescued': 1, 'ignored': 0}}
```

Comment:
this assertion would def. fail for ansible < 2.8 since the the new fields won't be surfaced at all on older versions

Reply:
doh, we're mocking the object, so the fields _will_ be set 👍